src/tests/Interop/Swift/SwiftErrorHandling/SwiftErrorHandling.cs

```diff
@@ -23,6 +23,15 @@ public class ErrorHandlingTests
     [DllImport(SwiftLib, EntryPoint = "$s18SwiftErrorHandling018conditionallyThrowB004willE0s5Int32VAE_tKF")]
     public static extern nint conditionallyThrowErrorOnStack(int willThrow, int dummy1, int dummy2, int dummy3, int dummy4, int dummy5, int dummy6, int dummy7, int dummy8, int dummy9, ref SwiftError error);
 
+    [UnmanagedCallConv(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    [DllImport(SwiftLib, EntryPoint = "$s18SwiftErrorHandling26nativeFunctionWithCallbackyyyyXEF")]
+    public static extern unsafe IntPtr NativeFunctionWithCallback(delegate* unmanaged[Swift]<SwiftError*, void> callback, SwiftError* error);
+
+    [UnmanagedCallersOnly(CallConvs = new Type[] { typeof(CallConvSwift) })]
+    private static unsafe void ProxyMethod(SwiftError* error) {
+        *error = new SwiftError((void*)42);
+    }
```

Comment:
I would also add some tests that have a return value too.

Reply:
Also, tests with more parameters. I expect that the ABI classification is not handled correctly (I assume it will think `error` is an actual parameter and allocate registers/stack space for it, messing up the following arguments).
A simplifying assumption for both RyuJIT and Mono might be that `SwiftError` is always last, but then we should put that in the documentation/design document (cc @jkoritzinsky @kotlarmilos)

Reply:
> I expect that the ABI classification is not handled correctly (I assume it will think error is an actual parameter and allocate registers/stack space for it, messing up the following arguments).

I'm not sure if this is the cleanest solution, but I did something akin to your support for `SwiftSelf` in reverse P/Invoke scenarios, where we initially say we're going to pass the `SwiftError*` parameter in `REG_SWIFT_ERROR` (I assume a Swift caller wouldn't use this register to pass an argument, as we intend to trash it anyway), but when generating the function prolog, we skip over the `SwiftError*` parameter so that it isn't given a register/stack space.

Reply:
That seems ok to me. Did you hit problems when not giving it any register/stack assignment at all? Since it isn't going to have any references I wouldn't expect too many problems around that.

The logic for `SwiftSelf` is needed for a different reason. I have a local change that generalizes `genFnCalleePrologRegArgs` to handle it and remove its special casing in the backend.

Reply:
> Did you hit problems when not giving it any register/stack assignment at all?

Yeah, the tests with return values/multiple args ran into what you described above:

> I expect that the ABI classification is not handled correctly (I assume it will think error is an actual parameter and allocate registers/stack space for it, messing up the following arguments).